packages/twenty-front/src/modules/settings/data-model/fields/forms/currency/hooks/useCurrencySettingsFormInitialValues.ts

```diff
@@ -0,0 +1,36 @@
+import { useFormContext } from 'react-hook-form';
+
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { CurrencyCode } from '@/object-record/record-field/types/CurrencyCode';
+import { SettingsDataModelFieldCurrencyFormValues } from '@/settings/data-model/fields/forms/currency/components/SettingsDataModelFieldCurrencyForm';
+import { applySimpleQuotesToString } from '~/utils/string/applySimpleQuotesToString';
+
+export const useCurrencySettingsFormInitialValues = ({
+  fieldMetadataItem,
+}: {
+  fieldMetadataItem?: Pick<FieldMetadataItem, 'defaultValue'>;
+}) => {
+  const initialAmountMicrosValue = null;
```

Comment:
why do we need to have the initialAmountMicrosValue as it does not matter in the Settings? is this for preview? i

Reply:
Actually I added `amountMicros` as a value in the form to simplify saving the default value in `SettingsObjectFieldEdit` and `SettingsObjectNewFieldStep2`. This allows the form values to be in the exact format we need and avoids having to check if the type is Currency to format the default value before saving.